File: C_Sharp_code/Example/WindowsFormsApp/CS_demo14/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace CS_demo14
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            Student stu = new Student();
            stu.setNo(101);
            stu.setName("Eason");
            stu.setMajor("Chinese");
            stu.info();
            Student stu2 = new Student();
            stu2.info();
            Student stu3 = new Student(102, "Rose", "Math");
            stu3.info();
        }
        public class Student
        {
            private static int _count;
            private int _no;
            private string _name;
            private string _major;
            public Student()
            {
                _no = -1;
                _name = "NA";
                _major = "NA";
            }
            public Student(int no, string name, string major)
            {
                this._no = no;
                this._name = name;
                this._major = major;
            }
            public int getCount()
            {
                return _count;
            }
            public void setCount()
            {
                _count++;
            }
            public int getNo()
            {
                return _no;
            }
            public void setNo(int no)
            {
                this._no = no;
            }
            public string getName()
            {
                return _name;
            }
            public void setName(string name)
            {
                this._name = name;
            }
            public string getMajor()
            {
                return _major;
            }
            public void setMajor(string major)
            {
                this._major = major;
            }
            public void info()
            {
                Console.WriteLine(_no + "," + _name + "," + _major);
            }
        }

        private void button2_Click(object sender, EventArgs e)
        {
            Student obj1 = new Student();
            obj1.setCount();
            Student obj2 = new Student();
            Console.WriteLine(obj1.getCount() + " " + obj2.getCount());
            obj2.setCount();
            Console.WriteLine(obj1.getCount() + " " + obj2.getCount());
        }
    }
}

File: C_Sharp_code/Example/WindowsFormsApp/VB_demo15/Form1.vb
Public Class Form1
    Public Class Student
        Private Shared _count As Integer
        Private _no As Integer
        Private _name As String
        Private _major As String
        '建構式
        Public Sub New()
            _no = -1
            _name = "NA"
            _major = "NA"
        End Sub
        Public Sub New(ByVal no As Integer, ByVal name As String,
                       ByVal major As String)
            Me._no = no
            Me._name = name
            Me._major = major
        End Sub
        Public Function getCount() As Integer
            Return _count
        End Function
        Public Sub setCount()
            _count += 1
        End Sub
        Public Function getNo() As Integer
            Return _no
        End Function
        Public Sub setNo(ByVal no As Integer)
            Me._no = no
        End Sub
        Public Function getName() As String
            Return _name
        End Function
        Public Sub setName(ByVal name As String)
            Me._name = name
        End Sub
        Public Function getMajor() As String
            Return _major
        End Function
        Public Sub setMajor(ByVal major As String)
            Me._major = major
        End Sub

        Public Sub info()
            Console.WriteLine(_no & "," & _name & "," & _major)
        End Sub
    End Class
    Private Sub button1_Click(sender As Object, e As EventArgs) Handles button1.Click
        Dim stu As New Student
        stu.setNo(102)
        stu.setName("Ken")
        stu.setMajor("English")
        stu.info()
        Dim stu2 As New Student()
        stu2.info()
        Dim stu3 As New Student(103, "Mary", "Math")
        stu3.info()
    End Sub

    Private Sub button2_Click(sender As Object, e As EventArgs) Handles button2.Click
        Dim obj1 As New Student
        obj1.setCount()
        Dim obj2 As New Student
        Console.WriteLine(obj1.getCount() & " " & obj2.getCount())
        obj2.setCount()
        Console.WriteLine(obj1.getCount() & " " & obj2.getCount())
    End Sub
End Class
